### DISCUSSION TITLE
Connection test feature have to run within Executor (instead of Webserver)

### DISCUSSION THREAD

**Discussion by pvaling** [apache/airflow#25988] [General] [answered]
### Description

There is a _connection test_ feature in Airflow UI. It behaves incorrect because real user tasks executed in Executor network environment and it may be different from Webserver process environment. For example In my deployment Webserver and Scheduler runs in a different k8s clusters with different network access policies. Webserver doesn`t have any access to datasources.

### Use case/motivation

Make connection test routines execute as real Executor jobs.

### Are you willing to submit a PR?

- [ ] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://github.com/apache/airflow/blob/main/CODE_OF_CONDUCT.md)

**Comment by potiuk** [comment, marked as answer]
See discussion here: https://github.com/apache/airflow/pull/15795 - we raised concern about this case and we decided that going with webserver-only is "good enough". And we knew there will be cases it won't work. 

I am not sure we are going to implement it as it is much more complex and the return of such investment make it not worth it in general. I would also personally like to get it done "right", but I also know that it involves complexities that are simply not worth for the "edge case" you have.

Let me convert it into discussion and maybe others who participated @msumit @kaxil changed their mind since. Also I encourage you @pvaling to take a look at the discuscion in #15795 and see what you get from it, and whether it changes your POV.

**Reply by pvaling** [reply]
Thank you for reply. 
Maybe we have to add test connection feature toggle into webserver config file to hide button from UI? Users are confused by Test button misleading behaviour. Solution looks dirty but will resolve such edge cases.
